### PR TITLE
[Updates]: fix VideoSlide ```typeStr```

### PR DESCRIPTION
**Pull request by gtalha07**
- adds serde tag support for ```NewsContent``` to strictly parse according to type and fallback type to support existing ones.
- add video slide support in rust integration tests.

### CODE REVIEW

File: native/acter/src/api/news.rs [hunk 1]
```diff
@@ -165,24 +166,49 @@ impl NewsSlide {
...
+            NewsContent::Video(VideoMessageEventContent { body, .. })
+            | NewsContent::Fallback(FallbackNewsContent::Video(VideoMessageEventContent {
+                body,
+                ..
+            })) => body.clone(),
+            NewsContent::Audio(AudioMessageEventContent { body, .. })
+            | NewsContent::Fallback(FallbackNewsContent::Audio(AudioMessageEventContent {
+                body,
+                ..
+            })) => body.clone(),
```
Comment (gnunicorn) [inline native/acter/src/api/news.rs:203]: aren't all these `body` of the same type? Could be one match in that case... but also we might want to consider moving that into `core` as it is a bit ... verbose... What do you think?

File: native/acter/src/api/news.rs [hunk 2]
```diff
@@ -194,12 +220,32 @@ impl NewsSlide {
...
+            }
+
+            NewsContent::Video(content)
+            | NewsContent::Fallback(FallbackNewsContent::Video(content)) => {
+                MsgContent::from(content)
+            }
+            NewsContent::Text(content)
+            | NewsContent::Fallback(FallbackNewsContent::Text(content)) => {
+                MsgContent::from(content)
+            }
```
Comment (gnunicorn) [inline native/acter/src/api/news.rs:248]: :+1:

File: native/core/src/events/news.rs [hunk 3]
```diff
@@ -30,61 +30,142 @@ pub enum NewsContent {
...
+    /// A text message.
+    Text(TextMessageEventContent),
+    /// An audio message.
+    Audio(AudioMessageEventContent),
+    /// A file message
+    File(FileMessageEventContent),
+    /// A location message.
+    Location(LocationMessageEventContent),
+
+    #[serde(untagged)]
```
Comment (gnunicorn) [inline native/core/src/events/news.rs:104]: Let's add some docs here:
```suggestion
    /// Backwards-compatible fallback support for previous untagged version
    /// only for reading existing events.
    #[serde(untagged)]
```

**Review by gnunicorn — APPROVED** [review]
lgtm. just some minor things inline we might want to consider

File: native/acter/src/api/news.rs (hunk 1 above)
Comment (gtalha07) [reply native/acter/src/api/news.rs:203]: Yes all cases have body as ```&String```. Could have possibility of moving in core, let me do it.

File: native/core/src/events/news.rs (hunk 3 above)
Comment (gtalha07) [reply native/core/src/events/news.rs:104]: Forgot it since you mentioned. Let me add!

**Pull request merged**